src/lib/server/cms.ts

```diff
@@ -29,6 +37,90 @@ purify.addHook('afterSanitizeAttributes', function (node) {
 		node.setAttribute('rel', 'noopener');
 	}
 });
+
+// Zod schema for external product data validation
+const externalProductSchema = z.object({
+	_id: z.string(),
+	name: z.string(),
+	shortDescription: z.string(),
+	price: z.object({
+		amount: z.number(),
+		currency: z.string()
+	}),
+	pictures: z.array(z.any()).optional()
+});
+
+async function fetchExternalProduct(url: string): Promise<ExternalProductData | null> {
+	try {
+		// Parse URL to extract base and slug
+		const urlObj = new URL(url);
+		const pathMatch = urlObj.pathname.match(/\/product\/([^\/]+)/);
+
+		if (!pathMatch) {
+			console.error('Invalid product URL format:', url);
+			return null;
+		}
+
+		const slug = pathMatch[1];
+		const apiUrl = `${urlObj.origin}/.well-known/product/${slug}`;
```

Comment:
Why is this on `.well-known` and not `/api`?

[RFC 8615](https://datatracker.ietf.org/doc/html/rfc8615) introduces Well-Known URIs as:

> the discovery of information about an origin

Here, we're not doing discovery nor interoperability with other “product-bearing” implementations. Furthermore, the same specifications clearly states:

> Applications that wish to mint new well-known URIs MUST register them, following the procedures in [Section 5.1](https://datatracker.ietf.org/doc/html/rfc8615#section-5.1) [...].